### PR TITLE
Better error message when removing examiner role from user account

### PR DESCRIPTION
**Pull request by cmadjar**
This pull request improves the error message when removing the examiner role without clearing the fields "Radiologist" and "Pending Approval" of the "Examiner Status" row. Removal of all examiner roles works but needs to have those fields cleared out as well.

See also: https://redmine.cbrain.mcgill.ca/issues/14716

### CODE REVIEW

**Comment by johnsaigle** [comment]
@cmadjar it looks like you've issued this PR to the wrong branch

**Comment by cmadjar** [comment]
@johnsaigle Thanks! Just fixed the branch the PR should go to.

File: modules/user_accounts/php/edit_user.class.inc [hunk 1]
```diff
@@ -1031,7 +1031,8 @@ class Edit_User extends \NDB_Form
                 || $values['examiner_pending'] !== '')
             ) {
                 $errors['examiner_sites'] = "Please select at least one examiner 
-                site.";
+                site or reset the 'Examiner status' fields blow (i.e.
```
Comment (nicolasbrossard) [inline modules/user_accounts/php/edit_user.class.inc]: ...fields *below*...

**Comment by cmadjar** [comment]
@nicolasbrossard Thanks! Just fixed the typo.

**Comment by ZainVirani** [comment]
are you sure that the removal of examiner fields is working?
On this candidate (https://test.loris.ca/user_accounts/edit_user/?identifier=slowloris) I clear the Examiner Status fields, and uncheck DCC, but when I hit save DCC is still checked

File: modules/user_accounts/php/edit_user.class.inc [hunk 2]
```diff
@@ -1031,7 +1031,8 @@ class Edit_User extends \NDB_Form
                 || $values['examiner_pending'] !== '')
             ) {
                 $errors['examiner_sites'] = "Please select at least one examiner 
-                site.";
+                site or reset the 'Examiner status' fields below (i.e. 
+                'Radiologist' and 'Pending Approval') to nothing.";
```
Comment (zaliqarosli) [inline modules/user_accounts/php/edit_user.class.inc]: could we say something like "clear the 'Examiner status' fields below ..."? to replace "to nothing"

**Comment by cmadjar** [comment]
> are you sure that the removal of examiner fields is working?
> On this candidate (https://test.loris.ca/user_accounts/edit_user/?identifier=slowloris) I clear the Examiner Status fields, and uncheck DCC, but when I hit save DCC is still checked

@ZainVirani I swear I thought the check box being emptied at saving. Wishful thinking I guess. This should be fixed now.

@ZainVirani @zaliqarosli I think I took care of all your comments. Thanks a lot!

**Comment by driusan** [comment]
@cmadjar This is failing due to PHPCS and phan.. it likely just needs to be rebased to fix the phan problem, and the phpcs should be easy enough to manually fix.

File: modules/user_accounts/php/edit_user.class.inc [hunk 3]
```diff
@@ -276,128 +276,124 @@ class Edit_User extends \NDB_Form
...
 
-            // first check if an update is needed then actually do it
-            if (!empty($ex_radiologist)
-                && !empty($ex_pending)
-                && !empty($ex_curr_sites)
-            ) {
-                //modify examiner radiologist to be in the binary format 0-1
-                $ex_radiologist = $ex_radiologist === 'Y' ? 1 : 0;
+            //modify examiner radiologist to be in the binary format 0-1
+            $ex_radiologist = $ex_radiologist === 'Y' ? 1 : 0;
```
Comment (driusan) [inline modules/user_accounts/php/edit_user.class.inc]: I always misread the variable "$ex_radiologist"..

**Pull request merged**